dotnet/src/Functions/Functions.OpenApi/OpenApi/OpenApiDocumentParser.cs

```diff
@@ -183,14 +186,63 @@ internal static List<RestApiOperation> CreateRestApiOperations(string? serverUrl
                 CreateRestApiOperationParameters(operationItem.OperationId, operationItem.Parameters),
                 CreateRestApiOperationPayload(operationItem.OperationId, operationItem.RequestBody),
                 CreateRestApiOperationExpectedResponses(operationItem.Responses).ToDictionary(item => item.Item1, item => item.Item2)
-            );
+            )
+            {
+                Extensions = CreateRestApiOperationExtensions(operationItem.Extensions)
+            };
 
             operations.Add(operation);
         }
 
         return operations;
     }
 
+    /// <summary>
+    /// Build a dictionary of extension key value pairs from the given open api extension model, where the key is the extension name
+    /// and the value is either the actual value in the case of primitive types like string, int, date, etc, or a json string in the
+    /// case of complex types.
+    /// </summary>
+    /// <param name="extensions">The dictionary of extension properties in the open api model.</param>
+    /// <returns>The dictionary of extension properties using a simplified model that doesn't use any open api models.</returns>
+    /// <exception cref="KernelException">Thrown when any extension data types are encountered that are not supported.</exception>
+    private static Dictionary<string, object?> CreateRestApiOperationExtensions(IDictionary<string, IOpenApiExtension> extensions)
+    {
+        var result = new Dictionary<string, object?>();
+
+        // Map each extension property.
+        foreach (var extension in extensions)
+        {
+            object? extensionValueObj = null;
+
+            if (extension.Value is IOpenApiPrimitive primitive)
+            {
+                // Set primitive values directly into the dictionary.
+                extensionValueObj = GetParameterValue(primitive, "extension property", extension.Key);
+            }
+            else if (extension.Value is IOpenApiAny any)
+            {
+                // Serialize complex objects and set as json strings.
+                // The only remaining type not referenced here is null, but the default value of extensionValueObj
+                // is null, so if we just continue that will handle the null case.
+                if (any.AnyType == AnyType.Array || any.AnyType == AnyType.Object)
+                {
+                    var schemaBuilder = new StringBuilder();
+                    var jsonWriter = new OpenApiJsonWriter(new StringWriter(schemaBuilder, CultureInfo.InvariantCulture), new OpenApiJsonWriterSettings() { Terse = true });
+                    extension.Value.Write(jsonWriter, Microsoft.OpenApi.OpenApiSpecVersion.OpenApi3_0);
+                    extensionValueObj = schemaBuilder.ToString();
+                }
+            }
+            else
+            {
+                throw new KernelException($"The type of extension property '{extension.Key}' is not supported while trying to consume the OpenApi schema.");
```

Comment:
I'm not sure if we need to throw an exception here. I'm considering cases when someone imports an OpenAPI specification that might have extensions like `OpenApiDeprecationExtension`, `OpenApiPagingExtension`, and other derivatives of `IOpenApiExtension`, but the scenario doesn't require those for execution. In those cases, we should not stop the import operation because we don't support the particular type of extension.  
   
On the other hand, there could be scenarios that require the extensions. To support them, they need to be reported, and corresponding functionality added.

Reply:
Maybe the best is just to avoid adding them to the collection if we do not recognize them. That way we can always add them later, once we have an appropriate way of representing them.